### ISSUE TITLE
non-integer `thin` values cause problems/warning

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#1249]
In my simple test, I see all 0s as the saved output. 

With `runMCMC`, it looks like we use `floor(thin)` and get a type conversion warning. See nimble-users email from 2022-11-21.

**Comment by danielturek** [comment]
@paciorek I'm not immediately seeing any case where "all samples returned are 0" (even when using `mcmcm$run`).  Do you have an example of that?

I can reproduce the warning.  I'm happy to address the warning.

**Comment by paciorek** [comment]
yeah, sorry for lack of reprex. 

```
code <- nimbleCode({
    y ~ dnorm(mu, sd = sigma)
    mu ~ dnorm(0,1)
    sigma ~ dunif(0,1)
})
m <- nimbleModel(code, inits = list(mu = 0, sigma = 1)
mcmc <- buildMCMC(m, thin = 1.3)
cm <- compileNimble(m)
cmcmc <- compileNimble(mcmc, project = m)
cmcmc$run(20)
as.matrix(cmcmc$mvSamples)
```

```
      mu sigma
 [1,] NA    NA
 [2,]  0     0
 [3,]  0     0
 [4,]  0     0
 [5,]  0     0
 [6,]  0     0
 [7,]  0     0
 [8,]  0     0
 [9,]  0     0
[10,]  0     0
[11,]  0     0
[12,]  0     0
[13,]  0     0
[14,]  0     0
[15,]  0     0
```

**Comment by danielturek** [comment]
Checks catching this case added in PR #1250

**Comment by paciorek** [comment]
Fixed by #1250, now merged in.
